src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -233,6 +233,9 @@ void ParticipantConfiguration::xmlTagCallback(
       throw std::runtime_error{stream.str()};
     }
     bool          provide = tag.getBooleanAttributeValue(ATTR_PROVIDE);
+    if (_participants.back()->getName() == from) {
+      PRECICE_CHECK(provide, "Participant \"" << context.name << "\" cannot use Mesh \"" << name << "\" from itself" << from << ". This is likely a typo in the from-field.");
```

Comment:
I am so sure about the typo. Normally participants have rather different names. If somebody runs into this I guess he/she did not understand what we mean with "from". 

```suggestion
      PRECICE_CHECK(provide, "Participant \"" << context.name << "\" cannot use Mesh \"" << name << "\" from itself" << from << ". With \"from\" you can mark a mesh that will be communicated from another participant. .");
```

Reply:
We could also give a hint on what to enter as `from`.
```suggestion
      PRECICE_CHECK(provide, "Participant \"" << context.name << "\" cannot use Mesh \"" << name << "\" from itself. Use \"from\" to mark a mesh provided by another participant.");                         
```

This argument should really not be necessary as the provider is explicitly configured.

Reply:
> This argument should really not be necessary as the provider is explicitly configured.

Yes, good point. This is more involved to get though. Actually, we only have two states, provided or received. So, the config could be changed (eventually) to 
```xml
<use-mesh name="MyMesh" source="provided/received" />
```
or a different name for "source". Would be a breaking change though.

Reply:
@uekerman Which version do you prefer?

Reply:
No strong opinion. I think some "communicated" in there helps to understand the concept.

Reply:
I rephrased the error message